### PR TITLE
:rocket: Feature: Public ShutdownWithContext

### PR DESCRIPTION
**Pull request by jcyamacho**
## Description

The Fiber app offers two shutdown options: `Shutdown` and `ShutdownWithTimeout`. Both of these options internally use a private `shutdownWithContext` method, which is more versatile for those using Fiber. By making `shutdownWithContext` public, we allow developers to create the best strategy for shutting down the app.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation - /docs/ directory for https://docs.gofiber.io/
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] If new dependencies exist, I have checked that they are really necessary and agreed with the maintainers/community (we want to have as few dependencies as possible)
- [x] I tried to make my code as fast as possible with as few allocations as possible
- [x] For new code I have written benchmarks so that they can be analyzed and improved

## Commit formatting:

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -860,11 +860,15 @@ func (app *App) Shutdown() error {
...
 }
 
-// shutdownWithContext shuts down the server including by force if the context's deadline is exceeded.
-func (app *App) shutdownWithContext(ctx context.Context) error {
+// ShutdownWithContext shuts down the server including by force if the context's deadline is exceeded.
+//
+// Make sure the program doesn't exit and waits instead for ShutdownWithTimeout to return.
+//
+// ShutdownWithContext does not close keepalive connections so its recommended to set ReadTimeout to something else than 0.
+func (app *App) ShutdownWithContext(ctx context.Context) error {
```
Comment (gaby) [inline app.go:871]: If the function is public, we may need to add a test.

**Review by gaby — CHANGES_REQUESTED** [review]
Now sure why the spacing was changed in the .md files

File: docs/api/app.md [hunk 2]
```diff
@@ -145,19 +145,19 @@ func (app *App) MountPath() string
 
 ```go title="Examples"
 func main() {
-	app := fiber.New()
```
[the remark this answers is not in the document]
Comment (jcyamacho) [reply docs/api/app.md]: thanks!! done

File: app.go (hunk 1 above)
Comment (jcyamacho) [reply app.go:871]: done! the test is almost equal to `Test_App_ShutdownWithTimeout` 😅

**Comment by jcyamacho** [comment]
> Now sure why the spacing was changed in the .md files

it was an auto-formatter in my local.. 😅

**Review by gaby — APPROVED** [review]
LGTM

**Review by efectn — APPROVED** [review]

**Pull request merged**